internal/broker_consumer/consumer.go

```diff
@@ -0,0 +1,141 @@
+package broker_consumer
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/golang/glog"
+	"github.com/openshift-hyperfleet/hyperfleet-broker/broker"
+)
+
+// Subscriber wraps the hyperfleet-broker Subscriber interface
+type Subscriber = broker.Subscriber
+
+// CloudEvent wraps the CloudEvents SDK Event type
+type CloudEvent = event.Event
+
+// HandlerFunc wraps the hyperfleet-broker HandlerFunc type
+type HandlerFunc = broker.HandlerFunc
+
+// isAlreadyExistsError checks if an error indicates a resource already exists
+func isAlreadyExistsError(err error) bool {
+	if err == nil {
+		return false
+	}
+	errMsg := strings.ToLower(err.Error())
+	return strings.Contains(errMsg, "already exists") ||
+		strings.Contains(errMsg, "alreadyexists") ||
+		strings.Contains(errMsg, "resource already exists")
+}
+
+// NewSubscriber creates a new subscriber from environment variables
+// using the hyperfleet-broker library.
+//
+// The subscriptionID parameter specifies which subscription/queue to consume from.
+// If empty, it will read from SUBSCRIPTION_ID or BROKER_SUBSCRIPTION_ID environment variable.
+//
+// Configuration is loaded from environment variables:
+//   - SUBSCRIPTION_ID or BROKER_SUBSCRIPTION_ID: subscription/queue name (if subscriptionID parameter is empty)
+//   - BROKER_TYPE: "rabbitmq" or "googlepubsub"
+//   - For RabbitMQ: BROKER_RABBITMQ_URL
+//   - For Google Pub/Sub: BROKER_GOOGLEPUBSUB_PROJECT_ID
+//   - SUBSCRIBER_PARALLELISM: number of parallel workers (default: 1)
+//
+// Example:
+//   sub, err := broker_consumer.NewSubscriber("my-subscription")
+//   if err != nil {
+//       log.Fatal(err)
+//   }
+//   defer sub.Close()
+//   
+//   handler := func(ctx context.Context, evt *event.Event) error {
+//       // Process the event
+//       return nil
+//   }
+//   
+//   err = sub.Subscribe(ctx, "my-topic", handler)
+
+func subscriptionIDFromEnv() string {
+	subscriptionID := os.Getenv("SUBSCRIPTION_ID")
+	if subscriptionID == "" {
+		subscriptionID = os.Getenv("BROKER_SUBSCRIPTION_ID")
```

Comment:
Can you point me to some docs around why we have two potential env vars for subscription ID and the same with Topic

Reply:
Removed the redundant one. We should follow [here](https://github.com/openshift-hyperfleet/architecture/blob/91a9b0795bb99ee1f7395fa89d478492c71d45bd/hyperfleet/components/adapter/framework/adapter-design-decisions.md?plain=1#L271)